Minor refactorings

karllindmark opened the pull request:
The loadImage() and setVisibility() were changes I've made for news - I could remove them from this PR and include them later on (if we don't keep them now).

@peter-budo

Commit:
Update network classes to be able to use specific headers per request

Commit:
Update adapter

Commit:
Pass Loader to onLoadSuccess() and onLoadFailure

TODO:
Deprecating current methods as they're used at many places
However, when no more onLoadSuccess(String) nor onLoadFailure(String)
implementations are left, we remove them below and make the

	onLoadX(Loader, String)

methods abstract.

Commit:
Remove abstract keyword from deprecated methods

peter-budo commented:
@karllindmark I will look into it today once back, need to do so shopping for Iceland trip first

peter-budo commented:
Build fails :-1: 

```
/Users/Peter/gradle-github/com.ninetwozero.battlelog/BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/fragments/AccountProfileFragment.java:38: cannot find symbol
symbol  : constructor ListRowAdapter(android.support.v4.app.FragmentActivity,java.util.List<com.ninetwozero.bf4intel.datatypes.ListRow>)
location: class com.ninetwozero.bf4intel.ui.adapters.ListRowAdapter
        final ListRowAdapter slidingMenuAdapter = new ListRowAdapter(getActivity(), getItems());
                                                  ^
/Users/Peter/gradle-github/com.ninetwozero.battlelog/BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/fragments/PlatoonProfileFragment.java:39: cannot find symbol
symbol  : constructor ListRowAdapter(android.support.v4.app.FragmentActivity,java.util.List<com.ninetwozero.bf4intel.datatypes.ListRow>)
location: class com.ninetwozero.bf4intel.ui.adapters.ListRowAdapter
        final ListRowAdapter slidingMenuAdapter = new ListRowAdapter(getActivity(), getItemsForMenu());
                                                  ^
/Users/Peter/gradle-github/com.ninetwozero.battlelog/BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/fragments/SoldierStatsFragment.java:39: cannot find symbol
symbol  : constructor ListRowAdapter(android.support.v4.app.FragmentActivity,java.util.List<com.ninetwozero.bf4intel.datatypes.ListRow>)
location: class com.ninetwozero.bf4intel.ui.adapters.ListRowAdapter
        final ListRowAdapter slidingMenuAdapter = new ListRowAdapter(getActivity(), getItemsForMenu());
```

peter-budo committed:
Fixed pull request

- Sorted issue with constructor parameters order
- Sorted issue with Gradle wrapper which was still 1.8 instead as project required 1.9
